rllib/algorithms/dt/dt.py

```diff
@@ -0,0 +1,361 @@
+import logging
+import math
+from typing import List, Optional, Type, Tuple, Dict, Any, Union
+
+from ray.rllib import SampleBatch
+from ray.rllib.algorithms.algorithm import Algorithm, AlgorithmConfig
+from ray.rllib.algorithms.dt.segmentation_buffer import MultiAgentSegmentationBuffer
+from ray.rllib.execution import synchronous_parallel_sample
+from ray.rllib.execution.train_ops import multi_gpu_train_one_step, train_one_step
+from ray.rllib.policy import Policy
+from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID
+from ray.rllib.utils.annotations import override, PublicAPI
+from ray.rllib.utils.metrics import (
+    NUM_AGENT_STEPS_SAMPLED,
+    NUM_ENV_STEPS_SAMPLED,
+    SAMPLE_TIMER,
+    NUM_AGENT_STEPS_TRAINED,
+)
+from ray.rllib.utils.typing import (
+    AlgorithmConfigDict,
+    ResultDict,
+    TensorStructType,
+    PolicyID,
+    TensorType,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class DTConfig(AlgorithmConfig):
+    def __init__(self, algo_class=None):
+        super().__init__(algo_class=algo_class or DT)
+
+        # fmt: off
+        # __sphinx_doc_begin__
+        # DT-specific settings.
+        # Required settings during training and evaluation:
+        # Initial return to go used as target during rollout.
+        self.target_return = None
+        # Rollout horizon/maximum episode length.
+        self.horizon = None
+
+        # Model settings:
+        self.model = {
+            # Transformer (GPT) context length.
+            "max_seq_len": 5,
+        }
+
+        # Transformer (GPT) settings:
+        self.embed_dim = 128
+        self.num_layers = 2
+        self.num_heads = 1
+        self.embed_pdrop = 0.1
+        self.resid_pdrop = 0.1
+        self.attn_pdrop = 0.1
+
+        # Optimization settings:
+        self.lr = 1e-4
+        self.lr_schedule = None
+        self.optimizer = {
+            # Weight decay for Adam optimizer.
+            "weight_decay": 1e-4,
+            # Betas for Adam optimizer.
+            "betas": (0.9, 0.95),
+        }
+        self.grad_clip = None
+
+        self.replay_buffer_config = {
+            # How many trajectories/episodes does the segmentation buffer hold.
+            # Increase for more data shuffling but increased memory usage.
+            "capacity": 20,
+            # Do not change the type of replay buffer.
+            "type": MultiAgentSegmentationBuffer,
+        }
+        # __sphinx_doc_end__
+        # fmt: on
+
+        # Overwriting the trainer config default
+        # If data ingestion/sample_time is slow, increase this
+        self.num_workers = 0
+        self.min_time_s_per_iteration = 10.0
```

Comment:
why 10 second per iteration? make it based on number of time steps. This will make the default algorithm none-deterministic.

Reply:
i dont really know, this is what CRR uses so I thought it was standard

Reply:
resolved. Changed to min_train_timesteps_per_iteration because evaluation_interval must be 1 to use evaluation/episode_reward_mean as the stopping condition

Reply:
sg.